app/client/lib/ethereum/observeBlocks.js

```diff
@@ -19,49 +19,64 @@ Update wallet balances
                 if(!err) {
                     // is of type wallet
-                    if(wallet.creationBlock) {
-                        Wallets.update(wallet._id, {$set: {
+                    if(account.creationBlock) {
+                        Wallets.update(account._id, {$set: {
                             balance: res.toString(10)
                         }});
                     } else {
-                        CustomContracts.update(wallet._id, {$set: {
+                        EthAccounts.update(account._id, {$set: {
+                            balance: res.toString(10)
+                        }});
+                        CustomContracts.update(account._id, {$set: {
                             balance: res.toString(10)
                         }});
                     }
                 }
             });
 
             // update dailylimit spent, etc, if wallet type
-            if(wallet.creationBlock) {
+            if(account.creationBlock) {
                 Meteor.setTimeout(function() {
-                    updateContractData(wallet);
+                    updateContractData(account);
                 }, 1000);
             }
         }
-    });
 
 
-    
-    // UPDATE TOKEN BALANCES
-    var walletsContractsAndAccounts = EthAccounts.find().fetch().concat(walletsAndContracts);
+        // Only check ENS names every N minutes
+        if (!account.ensCheck || (account.ensCheck && Date.now() - account.ensCheck > 10*60*1000)) {
+            Helpers.getENSName(account.address, function(err, name, returnedAddr) {
+                if (!err && account.address.toLowerCase() == returnedAddr){
+                    EthAccounts.update({address: account.address}, {$set:{ name: name, ens: true, ensCheck: Date.now()}});
+                    CustomContracts.update({address: account.address}, {$set:{ name: name, ens: true, ensCheck: Date.now()}});
+                    Wallets.update({address: account.address}, {$set:{ name: name, ens: true, ensCheck: Date.now()}});
+                } else {
+                    EthAccounts.update({address: account.address}, {$set:{ens: false, ensCheck: Date.now()}});
+                    CustomContracts.update({address: account.address}, {$set:{ens: false, ensCheck: Date.now()}});
+                    Wallets.update({address: account.address}, {$set:{ens: false, ensCheck: Date.now()}});
+
+                }
+            });
+        }
+
 
-    _.each(Tokens.find().fetch(), function(token){
-        if(!token.address)
-            return;
 
-        var tokenInstance = TokenContract.at(token.address);
+        _.each(Tokens.find().fetch(), function(token){
+            if(!token.address)
+                return;
+
+            var tokenInstance = TokenContract.at(token.address);
 
-        // go through all existing accounts, for each token
-        _.each(walletsContractsAndAccounts, function(account){
             tokenInstance.balanceOf(account.address, function(e, balance){
```

Comment:
With the changes above, i will also revert that, so we save on the custom contract address. Those don't need to be checked if they have tokens and would only create unnecessary queries.